specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,477 @@
+                });
+                return S_OK;
+            }).Get(),
+        &m_contextMenuRequestedToken);
+```
+## .Net/ WinRT Add or Remove Entries From Browser Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        CoreWebView2ContextMenuItemCollection collection = args.MenuItems;
+        CoreWebView2SelectionContext context = args.ContextSelected;
+        args.Handled = false;
+        if(context == CoreWebView2SelectionContext.Image)
+        {
+            /// removes the last item in the collection
+            collection.RemoveValueAtIndex(collection.Count - 1);
+        }
+        else if (context == CoreWebView2SelectionContext.Link)
+        {
+            /// add new item to end of colelction
+            CoreWebView2ContextMenuItem newItem = _coreWebView2Environment.CreateContextMenuItem(
+                "NewItem", "Shortcut");
+            collection.AddValueAtIndex(newItem, collection.Count);
+        }
+    };
+```
+
+## .Net/ WinRT Use Data to Display Custom Context Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        CoreWebView2Deferral deferral = args.GetDeferral();
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+            using (deferral)
+            {
+                IList<CoreWebView2ContextMenuItem> menuList = args.MenuItems;
+                CoreWebView2SelectionContext context = args.context;
+                args.Handled = true;
+                ContextMenu cm = this.FindResource("ContextMenu") as ContextMenu;
+                cm.Items.Clear();
+                for(int i = 0; i < menuList.Count; i ++){
+                    CoreWebView2ContextMenuItem current = menuList[i];
+                    MenuItem newItem = new MenuItem();
+                    newItem.Header = current.Name;
+                    newItem.Click += (s, ex) => args.SelectedItem = current.id;
+                    cm.Items.Add(newItem);
+                }
+                cm.IsOpen = true;
+            }
+        }, null);
+    };
+```
+# Remarks
+
+# API Notes
+
+# API Details
+ ```cpp
+    interface ICoreWebView2_4;
+    interface ICoreWebView2Environment;
+    interface ICoreWebView2ContextMenuItem;
+    interface ICoreWebView2ContextMenuItemCollection;
+    interface ICoreWebView2ContextMenuRequestedEventArgs;
+    interface ICoreWebView2ContextMenuRequestedEventHandler;
+
+    /// Defines the Context Menu Items
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR {
+        /// Custome type, will include Items brought in by browser extensions
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUSTOM,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_MISSPELLED,
+
+        /// No Command
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_NONE,
+
+        /// Page Commands
```

Comment:
We have a tool that generates our reference documentation. /// comments are reference documentation for the next thing so these sort of grouping / heading docs won't work.